modules/git/repo.go

```diff
@@ -80,19 +80,12 @@ func InitRepository(ctx context.Context, repoPath string, bare bool) error {
 // IsEmpty Check if repository is empty.
 func (repo *Repository) IsEmpty() (bool, error) {
 	var errbuf strings.Builder
-	if err := NewCommandContext(repo.Ctx, "log", "-1").RunInDirPipeline(repo.Path, nil, &errbuf); err != nil {
-		defaultBranch, err := repo.GetDefaultBranch()
-		if err != nil {
-			return false, err
-		}
-		if strings.Contains(errbuf.String(), "fatal: bad default revision 'HEAD'") ||
-			strings.Contains(errbuf.String(), fmt.Sprintf("fatal: your current branch '%s' does not have any commits yet", defaultBranch)) {
-			return true, nil
-		}
+	if err := NewCommandContext(repo.Ctx, "rev-list", "--all", "--count", "--max-count=1").RunInDirPipeline(repo.Path, nil, &errbuf); err != nil {
```

Comment:
The rev-list count is returned on stdout, so also need to receive that:

add `var resbuf strings.Builder`

and change line 83 to `if err := NewCommandContext(repo.Ctx, "rev-list", "--all", "--count", "--max-count=1").RunInDirPipeline(repo.Path, &resbuf, &errbuf); err != nil {`

also the output has a newline which needs to be trimmed to avoid Atoi erroring
thus line 87 needs to become `c, _ := strconv.Atoi(strings.TrimSpace(resbuf.String()))`

may also want to check the error on the conversion as well

this passes the mirror pull integration test and the repoisempty test

Reply:
Good catch